test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -606,7 +651,7 @@ public class HubEndpointTests
 
             using (var client = new TestClient())
             {
-                var endPointTask = endPoint.OnConnectedAsync(client.Connection);
+                Task endPointTask = await client.ConnectAsync(endPoint);
```

Comment:
Help me understand all of these test changes.

Reply:
`ConnectAsync` does 3 things:
1. Sends negotiate request message (previously done in TestClient ctor 💩 )
2. Calls `OnConnectedAsync`
3. Reads negotiate response message

Reply:
Where is the endpoint task? Do we still return it (unawaited?)

Reply:
Yes, `ConnectAsync` is `Task<Task>` and the inner task is the endpoint task.

Reply:
I'd never do it on a public API, it is too easy to not `await` and get the task result, but it works here for testing.